cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -842,6 +833,97 @@ func deleteDeviceModelAndVisitors(deviceModel *v1alpha2.DeviceModel, deviceProfi
 	}
 }
 
+func (dc *DownstreamController) sendDeviceMsg(device *v1alpha2.Device, operation string) {
+	device.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "devices.kubeedge.io",
+		Version: "v1alpha2",
+		Kind:    "Device",
+	})
+	modelMsg := model.NewMessage("").
+		SetResourceVersion(device.ResourceVersion).
+		FillBody(device)
+	modelResource, err := messagelayer.BuildResource(
+		device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0],
+		device.Namespace,
+		constants.ResourceTypeDevice,
+		device.Name)
+	if err != nil {
+		klog.Warningf("Built message resource failed for device, device: %s, operation: %s, error: %s", device.Name, operation, err)
+		return
+	}
+
+	// filter operation
+	switch operation {
+	case model.InsertOperation:
+	case model.DeleteOperation:
+	case model.UpdateOperation:
+	default:
+		klog.Warningf("unknown operation %s for device %s when send device msg", operation, device.Name)
+		return
+	}
+	modelMsg.BuildRouter(modules.DeviceControllerModuleName, "resource", modelResource, operation)
+
+	err = dc.messageLayer.Send(*modelMsg)
+	if err != nil {
+		klog.Errorf("Failed to send device addition message %v, device: %s, operation: %s, error: %v",
+			modelMsg, device.Name, operation, err)
+	}
+}
+
+func (dc *DownstreamController) sendDeviceModelMsg(device *v1alpha2.Device, operation string) {
```

Comment:
I do not see the releted logic when the device model is updated, the device model info send to the node

Reply:
we can have a cache to store the node to the device model, like configmap to pod, otherwise if the device model is updated, the info may take a long time to send to the mapper